sycl/doc/extensions/Matrix/dpcpp-joint-matrix.asciidoc

```diff
@@ -0,0 +1,223 @@
+The base pointer determines the starting address of the matrix to be loaded/stored.
+`layout` determines whether the data are being read/written in a row (`row_major`), column major (`column_major`) fashion, or if the data has already been transformed into VNNI format (`packed_a`, `packed_b`). 
+
+```c++
+void matrix_load(matrix<>A, T *base, unsigned stride, matrix_layout l = row_major);  
+```
+This function loads data from memory to the 2d tiles of AMX that is a 2d storage.
+```c++
+void matrix_store(matrix<>A, T *base, unsigned stride, matrix_layout l = row_major); 
+```
+This function stores the data from the 2d tiles back to memory.
+```c++
+matrix<> matrix_mad(matrix<>A, matrix<>B, matrix<>C);
+```
+
+The matrix multiply and add function performs the multiply operation on the matrices `A` and `B`, accumulate the result with `C` and return the result.
+
+## Integration with DPC++
+When using the matrix interface in a DPC++ kernel, additional semantics have to be added to define the memory scope of the matrix object and the execution scope of its operations. In the context of DPC++, `matrix` is distributed among an execution unit. In practice this can be one work-item, the work-items in a sub-group, or the work-items in a work-group.
+
+Since the matrix functions are group operations (as defined in Section 4.17.3 of the SYCL specification), the matrix API has to be accessed by all the work-items in the group in a non-diverged control flow. The `Group` template argument can be a work-group or a sub-group. These functions will be called once by all the work items in a group.
+
+### Memory Scope
+For the memory scope, we have two solutions. The long term solution is to use the proposed https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/LocalMemory/SYCL_INTEL_local_memory.asciidoc[`group_local_memory` extension] to allocate the matrix in local memory associated with a SYCL group as shown in the example below.
+
+```c++
+multi_ptr<matrix<T>, address_space::local_space> tA_ptr = group_local_memory<matrix<sub_group, int8_t, tM, tN>>(sg);
+```
+However, sub-group local memory is not yet well defined in DPC++. Moreover, the representation of this notion in LLVM IR and SPIR-V is not yet clear. Hence, for this proposal, we will proceed with adding the memory scope as an additional constructor argument as follows:
+
+```c++
+joint_matrix<sub_group, int8_t, tM, tN> tA(sg); 
+```
+### Execution Scope
+To be aligned with the SYCL 2020 group algorithms, an additional group argument is added to the matrix operations to designate that these functions are collective operations. The DPC++ syntax is the following: 
+
+```c++
+void joint_matrix_load(Group g, matrix<>A, T *base, unsigned stride, matrix_layout l = row_major);  
+```
+```c++
+void joint_matrix_store(Group g, matrix<>A, T *base, unsigned stride, matrix_layout l = row_major); 
+```
+```c++
+matrix<> joint_matrix_mad(Group g, matrix<>A, matrix<>B, matrix<>C);
+```
```

Comment:
General comments on these functions:

* It would be better to show a synopsis that included the namespace (similar to my comment above about the `matrix` type.

* It would also be better to include the full template declaration.  What you have now is not really C++ syntax.  This would help clarify, for example, whether any of the template parameters for A, B, and C need to match in `joint_matrix_mad`.

Reply:
I will incorporate this and the rest to the revision of the document